copy/copy.go

```diff
@@ -816,7 +926,32 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 	// Note that for this check we don't use the stronger "validationSucceeded" indicator, because
 	// dest.PutBlob may detect that the layer already exists, in which case we don't
 	// read stream to the end, and validation does not happen.
-	digestingReader, err := newDigestingReader(srcStream, srcInfo.Digest)
+
+	var decrypted bool
+	var err error
+	if manifest.IsEncryptedLayer(srcInfo) {
```

Comment:
This can’t copy encrypted layers unmodified from one registry to another?

Reply:
We've just added support for this. You can now copy images that are encrypted to another registry (remaining encrypted). Providing decryption keys allows decryption as well if one would like to decrypt the image. We will add some examples to the Skopeo tests.